manifests/init.pp

```diff
@@ -216,23 +216,33 @@
     # Set up Hiera. Even though it's not necessary to explicitly set $type for the authoritative
     # config, it is added for clarity.
     $powerdns_auth_config = lookup('powerdns::auth::config', Hash, 'deep', {})
-    $powerdns_auth_defaults = { 'type' => 'authoritative' }
-    create_resources(powerdns::config, $powerdns_auth_config, $powerdns_auth_defaults)
+    $powerdns_auth_config.each |$conf_title, $data| {
+      powerdns::config { $conf_title:
+        * => $data + { 'type' => 'authoritative' },
```

Comment:
Shouldn't `$data` override the defaults?

```suggestion
        * => { 'type' => 'authoritative' } + $data,
```

Reply:
The third argument to `create_resources` is the defaults hash: https://help.puppet.com/core/8/Content/PuppetCore/Markdown/function.htm#create_resources